Add Layer info data as a layer attribute

hanbyul-here opened the pull request:
Part of #823 

**Heavily edited after the last time I opened this PR**

With this pr, layer can include `info` attribute. `info` consists of
```
info:
  modal: 
    subtitle: |
      ::markdown subtitle in markdown
    contents: |
      ::markdown contents in markdown
  template:
        source: string
        spatialExtent: string
        latency: string
        unit: string
``` 

I added a sandbox page to trigger the modal.   https://deploy-preview-843--veda-ui.netlify.app/sandbox/sandboxLayerInfo 

@faustoperez Currently, there is no easy way to determine if the link the user uses is an external or an internal link. (The link is just part of the HTML that a user passes as a part of Layerdata.) So I think we might have to skip the external link icons. :[ 


I am not implementing the link on top of the header, since I can pick that component up from the other branch currently working on (https://github.com/NASA-IMPACT/veda-ui/pull/829) 
![Screen Shot 2024-02-14 at 10 48 14 AM](https://github.com/NASA-IMPACT/veda-ui/assets/4583806/f0928050-bc3a-4391-aed8-ddd1c57fd5d4)


**Details of technical implementation**: This PR required an Implementation of processing markdown in frontmatter in `stringfy-yml-func`.
- I thought about passing Markdown as it is and delegating the rendering of it to the front-end, but it will increase the load of front-end work at this point. (Integrating it with our current MDX system for rendering stories and datasets seems challenging.) 
- Initially considered using our system's markdown parsers ([remark](https://github.com/remarkjs/remark) and [rehype](https://github.com/rehypejs/rehype)), but they are pure ES modules requiring rewiring in the pipeline, which I wanted to avoid in this PR. Therefore, I chose a different markdown parser.
- It's important to note that `info.modal.subtitle` and `info.modal.contents` should only contain markdown, not MDX, to exclude custom components.

j08lue commented on parcel-resolver-veda/index.d.ts:
```diff
@@ -56,11 +56,23 @@ declare module 'veda' {
+  export interface LayerFrontmatterInfo {
+    info: {
+      description: string;
+      table: {
+        source: string;
+        spatialExtent: string;
+        latency: string;
+        unit: string;
```
I guess this is the Sandbox thing, right? Could it basically just take whatever is in the config object and render that, rather than hard-coding keys?

hanbyul-here replied:
Yes it can. But that approach can be error-prone, and we do need to have at least these 4 to display at the layer level on the timeline if I am understanding the design right (unless we are fine with the data being irregular)

j08lue commented on mock/datasets/no2.data.mdx:
```diff
@@ -55,6 +55,20 @@ layers:
+
+        [A tag test](https://developmentseed.org)
+      table:
+        source: NASA
+        spatialExtent: Global
+        latency: Monthly
+        unit: unit
```
We will want to have nicer names for the keys - capitalized, with spaces and possibly unicode chars, etc.

```suggestion
        Source: NASA
        Spatial extent: Global
        Latency: Monthly
        Units: 10¹⁵ molecules cm⁻²
```

If that does not work nicely with the current object structure in YAML, perhaps we can use something like this?

```yaml
        - title: Source
          value: NASA
        - title: Spatial extent
          value: Global
```

or

```yaml
        source:
          title: Source
          value: NASA
        spatialExtent: 
          title: Spatial extent
          value: Global
```

hanbyul-here replied:
This change is fine but to clarify my understanding: 
Maybe my sandbox example was confusing (that the example renders the key name) from the current design, I only see this table data will be used programmatically (key will never be rendered directly, and will be used only to fetch the data from the layer.) Considering we offer description as markdown (which this information will be included as markdown list), do we need to make key name human-friendly?

j08lue reviewed (changes requested):
We need nice property titles, not just slug-like keys, please.

hanbyul-here commented:
I think I poorly communicated what the code does in this PR. I will close this for now and open it again after making it clear.

j08lue reviewed (approved):
Looks good and the choice of making it all Markdown is solid for now (see comment below).

![image](https://github.com/NASA-IMPACT/veda-ui/assets/3404817/f36db968-8505-424a-bd7e-b7bd88c40789)

hanbyul-here commented on mock/datasets/no2.data.mdx:
```diff
@@ -55,6 +55,25 @@ layers:
     description: Levels in 10¹⁵ molecules cm⁻². Darker colors indicate higher nitrogen dioxide (NO₂) levels associated and more activity. Lighter colors indicate lower levels of NO₂ and less activity.
+    info:
+      modal:
+        subtitle: |
+          ::markdown
+          Global, monthly 1 km resolution dataset of fossil fuel carbon dioxide emissions, version 2022
+        contents: |
+          ::markdown
+          - Temporal Extent: January 2000 - December 2021
+          - Temporal Resolution: Monthly
+          - Spatial Extent: Global
+          - Spatial Resolution: 1 km x 1 km
+          - Data Units: Tons of carbon per 1 km x 1 km cell (monthly total)
+          - Data Type: Research
+          - Data Latency: Updated annually, following the release of an updated [BP Statistical Review of World Energy report](https://www.bp.com/en/global/corporate/energy-economics.html)
```
Yeah, I agree. Especially since our current application also asks for structured data (why `info` has a separate section for `template`). We would need more streamlined thoughts about how to handle this layer information before though. I can't think of a straightforward way of having both descriptive layerinfo and structured layerinfo without manual edits and duplication of the data.

sandrahoang686 commented on app/scripts/components/sandbox/layer-info/index.js:
```diff
@@ -0,0 +1,47 @@
+      </Block>
+    </>
+  ) : (
+    <div> Cannot find layer info fron dataset no2</div>
```
🔧 : Mispelled `from` and should `no2` be hardcoded here?

hanbyul-here commented:
I moved markdown level data to the dataset (`infoDescription` - please let me know if there is a better way to name these.), left `info` to layer level. Added these two to documentation ticket: https://github.com/NASA-IMPACT/veda-ui/issues/842
